File: Betacycle BackEnd Aggiornato 20-05/BetaCycle/Models/Payment.cs
using System;
using System.Collections.Generic;

namespace BetaCycle.Models;

public partial class Payment
{
    public long IdPayment { get; set; }

    public string NumberCard { get; set; } = null!;

    public string NumberCardSalt { get; set; } = null!;

    public string Cvv { get; set; } = null!;

    public string Cvvsalt { get; set; } = null!;

    public DateOnly ExpirationDate { get; set; }

    public string NameCard { get; set; } = null!;

    public string SurnameCard { get; set; } = null!;

    public string CircuitCard { get; set; } = null!;

    public long UserId { get; set; }

    public virtual Credential User { get; set; } = null!;
}

File: BetacycleFrontend/src/app/shared/models/payment.ts
/** Class used for the payment info of the user
 * @param idPayment Id of the Payment number
 * @param numberCard Number of the creditcard
 * @param numberCarSalt Salt of the number that will be addend in backend
 * @param cVV CVV of the credit card
 * @param cVVSalt Salt of the cvv that will be addend in backen
 * @param name First name on the card
 * @param surnameCard Surname on the card
 * @param circuitCard Circuit of the credit card
 * @param expirationDate credit card expiration date
 * @param userId Id from the User
 *  */
export class Payment{
    idPayment:number=0
    numberCard:string=''
    numberCarSalt:string=''
    cVV: string = ''
    cVVSalt: string = ''
    nameCard:string=''
    surnameCard:string=''
    circuitCard:string=''
    expirationDate:string=''
    userId:number=0
}